src/errors.c

```diff
@@ -95,19 +95,23 @@ void git_error_vset(int error_class, const char *fmt, va_list ap)
 		set_error_from_buffer(error_class);
 }
 
-void git_error_set_str(int error_class, const char *string)
+int git_error_set_str(int error_class, const char *string)
 {
 	git_buf *buf = &GIT_GLOBAL->error_buf;
 
 	assert(string);
 
-	if (!string)
-		return;
+	if (!string) {
+		git_error_set(GIT_ERROR_INVALID, "unspecified caller error");
+		return -1;
+	}
 
 	git_buf_clear(buf);
 	git_buf_puts(buf, string);
 	if (!git_buf_oom(buf))
```

Comment:
Should it return an error then `git_buf_oom` is true?

Reply:
Yeah, I think that you're right.  I was sort of thinking that callers wouldn't really need to know when there was an OOM situation, as they were trampolining out of this anyway and there's very little that they can do when we've run out of memory.  But I don't really see a reason why we shouldn't.  Updated to return `-1`.